src/cache_client.rs

```diff
@@ -289,6 +290,116 @@ impl CacheClient {
         request.send(self).await
     }
 
+    /// Fetches a dictionary from a cache.
+    ///
+    /// # Arguments
+    ///
+    /// * `cache_name` - The name of the cache containing the dictionary.
+    /// * `dictionary_name` - The name of the dictionary to fetch.
+    ///
+    /// # Examples
+    /// Assumes that a CacheClient named `cache_client` has been created and is available.
+    /// ```
+    /// # fn main() -> anyhow::Result<()> {
+    /// # use std::collections::HashMap;
+    /// # use std::convert::TryInto;
+    /// # use momento_test_util::create_doctest_cache_client;
+    /// # tokio_test::block_on(async {
+    /// use momento::cache::{DictionaryFetchRequest, DictionaryFetch};
+    /// # let (cache_client, cache_name) = create_doctest_cache_client();
+    /// let dictionary_name = "dictionary";
+    ///
+    /// let set_response = cache_client.dictionary_set_field(
+    ///   cache_name.to_string(),
+    ///   dictionary_name,
+    ///   "field1",
+    ///   "value1"
+    /// ).await?;
+    ///
+    /// let fetch_response = cache_client.dictionary_fetch(cache_name, dictionary_name).await?;
+    ///
+    /// match fetch_response {
+    ///    DictionaryFetch::Hit{ value } => {
+    ///       let dictionary: HashMap<String, String> = value.try_into().expect("I stored a dictionary!");
+    ///       println!("Fetched dictionary: {:?}", dictionary);
+    ///    }
+    ///    DictionaryFetch::Miss => println!("Cache miss"),
+    /// }
+    /// # Ok(())
+    /// # })
+    /// # }
+    /// ```
+    /// You can also use the [send_request](CacheClient::send_request) method to fetch a dictionary using a [DictionaryFetchRequest].
+    pub async fn dictionary_fetch(
+        &self,
+        cache_name: impl Into<String>,
+        dictionary_name: impl IntoBytes,
+    ) -> MomentoResult<DictionaryFetch> {
+        let request = DictionaryFetchRequest::new(cache_name, dictionary_name);
+        request.send(self).await
+    }
+    // dictionary_get_field
+    // dictionary_get_fields
+    // dictionary_increment
+    // dictionary_length
+    // dictionary_remove_field
+    // dictionary_remove_fields
```

Comment:
since we're aiming to implement just a foundational set of APIs for each collection type, should we be prioritizing the ones that operate on one field at a time or on multiple fields?

just thinking of how best to coordinate since I'm working on lists at the moment and had opted to prioritize the concatenate front/back rather than push front/back methods

Reply:
These were just placeholder comments for $futureSelf or $futureDev as to what commands are left. 👍 

> just thinking of how best to coordinate since I'm working on lists at the moment and had opted to prioritize the concatenate front/back rather than push front/back methods
Sounds good to me though you can coordinate with @cprice404 .